### PR TITLE
Log step time after every training step

### PR DESCRIPTION
**Pull request by ghost**
Logs step time after every training step. This is necessary in order for us to compare simulated step time with measured step time.

Example:

```
2024-01-25T22:35:02.248850019Z I0125 22:35:02.248594 133383564818176 entrypoint_utils.py:183] step_time: 5.806296268 s 
2024-01-25T22:35:08.079717010Z I0125 22:35:08.079638 133383564818176 entrypoint_utils.py:183] step_time: 5.830592737 s
2024-01-25T22:35:13.898919305Z I0125 22:35:13.898839 133383564818176 entrypoint_utils.py:183] step_time: 5.81876754 s
2024-01-25T22:35:19.722401602Z I0125 22:35:19.722332 133383564818176 entrypoint_utils.py:183] step_time: 5.822965802 s
2024-01-25T22:35:25.544759516Z I0125 22:35:25.544670 133383564818176 entrypoint_utils.py:183] step_time: 5.820692331 s
```

### CODE REVIEW

File: nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py [hunk 1]
```diff
@@ -534,6 +539,33 @@ def initialize_ub_func(self):
         )
         self.initialize_ub = False
 
+    def on_train_batch_start(self, batch, batch_idx):
+        super().on_train_batch_start(batch, batch_idx)
+        P = parallel_state.get_pipeline_model_parallel_rank()
+        T = parallel_state.get_tensor_model_parallel_rank()
+        D = parallel_state.get_data_parallel_rank()
+        R = torch.distributed.get_rank()
```
Comment (gkroiz) [inline nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py]: What is the purpose of creating these constants?

File: nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py [hunk 2]
```diff
@@ -534,6 +539,33 @@ def initialize_ub_func(self):
...
+        nvtx.end_range(self._nvtx_range)
+
+        if self.timestamp:
+            now = time.time_ns()
+            self.log_step_time(now - self.timestamp)
+            self.timestamp = None
+
+    def log_step_time(self, step_time_ns):
+        step_time = step_time_ns / 1000000000
+        instance_zone = "unknown"
```
Comment (gkroiz) [inline nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py]: what is `instance_zone` for?

File: nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py (hunk 1 above)
Comment (ghost) [reply nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py]: Replaced them with more clearly named local variables. They are used in the NVTX range annotation below.

File: nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py (hunk 2 above)
Comment (ghost) [reply nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py]: Deleted, an artifact from the original patch code I got from Patrick.

**Comment by crankshaw-google** [comment]
LGTM. Will wait for @gkroiz for approval.

**Review by gkroiz — APPROVED** [review]
LGTM

**Pull request merged**